libwallet/assets_manager.go

```diff
@@ -992,11 +1008,82 @@ func (mgr *AssetsManager) PrepareDexSupportForDcrWallet() error {
 			return nil, fmt.Errorf("DEX wallet not supported for %s", walletParams.Name)
 		}
 
-		return dcr.NewDEXWallet(dcrAsset.Wallet, dcrAsset.SyncData()), nil
+		return dcr.NewDEXWallet(dcrAsset, accountNumber, dcrAsset.SyncData()), nil
 	}
 
 	err := dexDcr.RegisterCustomWallet(walletMaker, def)
-	if err != nil && !strings.Contains(err.Error(), "already support" /* this is part of the error string returned if we've already registered a customer wallet */) {
+	if err != nil && !errors.Is(err, dexasset.ErrWalletTypeAlreadyRegistered) { // this is the only error from RegisterCustomWallet, ignore?
+		return err
+	}
+
+	return nil
+}
+
+// PrepareDexSupportForBTCWallet sets up the DEX client to allow using a
+// Cyptopower btc wallet with DEX core.
+func (mgr *AssetsManager) PrepareDexSupportForBTCWallet() error {
+	// Build a custom wallet definition with custom config options for use by
+	// the dexbtc.ExchangeWalletSPV.
+	customWalletConfigOpts := []*asset.ConfigOption{
+		{
+			Key:         dexc.WalletIDConfigKey,
+			DisplayName: "Wallet ID",
+			Description: "ID of existing wallet to use",
+		},
+		{
+			Key:         dexc.WalletAccountNumberConfigKey,
+			DisplayName: "Wallet Account Number",
+			Description: "Account number of the selected wallet",
+		},
+	}
+
+	def := &asset.WalletDefinition{
+		Type:        dexc.CustomDexWalletType,
+		Description: "Uses an existing cryptopower Wallet instance instead of an rpc connection.",
```

Comment:
There are now other alternative options aside rpc wallets, so just don't mention the alternatives.
```suggestion
		Description: "Uses an existing cryptopower Wallet.",
```